🔨 Scrape Number and Specify Description

migbro opened the pull request:
Partly addresses feature request # https://github.com/brentp/echtvar/issues/33
 - Is able to parse and return the correct `Number` attribute for info to update_header
 - Allows user to specify in config a `Description` to override the default
 TO DO: Actually scrape the description from the source vcf file

Built locally, got compiler warning:
```
warning: unreachable pattern
   --> src/commands/encoder_cmd.rs:226:13
    |
226 |             _ => panic!(
    |             ^
    |
    = note: `#[warn(unreachable_patterns)]` on by default
```
When run, headers are output as desired

brentp commented on src/lib/echtvar.rs:
```diff
@@ -177,16 +177,21 @@ impl EchtVars {
+                    if &e.description.to_string() == "added by echtvar"{
+                        format!("added by echtvar from {}", path)
+                    } else {
+                        e.description.to_string()
```
I think it would also be useful to prefix this with "added by echtvar"

brentp commented on Cargo.toml:
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "echtvar"
-version = "0.1.9"
+version = "0.2.0-rc"
```
can you revert this version change? I will bump the version after merging the PR.

brentp reviewed (changes requested):
Thanks very much! Looks good. 
Can you also add a test?

brentp commented on src/lib/echtvar.rs:
```diff
@@ -177,16 +177,21 @@ impl EchtVars {
             header.push_record(
                 format!(
-                    "##INFO=<ID={},Number=1,Type={},Description=\"{}\">",
+                    "##INFO=<ID={},Number={},Type={},Description=\"{}\">",
```
It's been too long since I looked at echtvar, but I think in fact, all fields should have `Number=1` since echtvar requires things to be decomposed. Thoughts?

migbro replied:
So, I have an instance where, since my team wants all gnomAD variants annotated, even the non-PASS ones, I Am adding a custom field in `INFO` with the gnomAD `FILTER` value so that the user can know whether it was a gnomAD `PASS` or not. There can be more than one possible fail reason, so that's why in this case `.` is possible. Aside from that, I definitely see your point in that one value per variant is likely to be the result of decomposing your reference.

brentp replied:
I think even for the case you describe, you can have Number=1,Type=String and have comma-delimited, right?

dmiller15 replied:
While it takes care of `A,R,G` types, decomposition doesn't necessarily preclude records from having fields with multiple values. I guess the real danger case here is an input of `Number=.,Type=Float`. I don't think the float conversion can handle the commas, and, while it can handle it, the string conversion is potentially looking at an enormous lookup and a confusing header `Number=1,Type=String,Description="Comma separated floats for something"`.

Additionally, do you know if setting the number to `1` would affect bcftools' ability to parse the value using its expressions? With `Number=.` you can specify the position in expressions like 'INFO/GNOMAD_FILTER[3] == "something"'`. Admittedly vague usefulness here but I've come across multi-value fields where performing position specific lookups is necessary.

migbro replied:
Assuming @dmiller15 is right, I have updated this to have A R G default to 1 and to leave the rest as-is

brentp commented on tests/big.sh:
```diff
@@ -61,7 +61,7 @@ for mod in 2 3 4 5; do
 done
-rm generated-all.vcf generated-subset0.vcf anno.vcf.gz test.echtvar0 test.echtvar1
+# rm generated-all.vcf generated-subset0.vcf anno.vcf.gz test.echtvar0 test.echtvar1
 bash string.sh
```
why did you change this?

migbro replied:
Well, I am not ready for  re-review yet. I made the change locally and I am using a remote machine to test. I hadn't made my test script yet, but just wanted to see the outputs in action before they are removed. I guess it's a little awkward to work on my own branch under scrutiny (I don't do cross-repo PRs often with external collaborators), but I'll re-request review when I think it's ready.

migbro commented:
@brentp , so I have added the test, but in the process I was thinking, what if instead of prepending `added by echtvar` to all the INFO fields when one is provided or scraped (in the future), I instead do something similar like add an echtvar command line, like bcftools and VEP does. It's part of the second commit of https://github.com/brentp/echtvar/pull/34/files/580fb12830f55693dc0dd7d2165e00b9cfacaa91..3c4eca58fca3060d037059ed8083f958b9c26842 and when I tested it out, it came out like:
`##echtvar_anno_Command=anno -i None generated-all.vcf anno.vcf.gz -e ["test.echtvar0"]`
I can mess around with string a bit ( I am a complete rust novice ). Thoughts? Thanks in advance for your attentiveness!

brentp commented:
@migbro , nicely done. I am OK with that echtvar command string. You can just [join](https://doc.rust-lang.org/std/primitive.slice.html#method.join) the archives with " -e " to get rid of the square brackets.

Let me know when you're ready for review.

migbro commented:
@brentp ok, I think I have addressed PR comments and suggestions, since the last look I have:
 - Cleaned up the added command line
 - Dropped "added by echtvar" for custom description fields only
 - Convert A R G to 1, for `Number`, leave all other values alone
 - Updated tests to account for changes

brentp commented:
thank you! 
I'll get this out in next release with changes from @dmiller15 in #35
